### ISSUE TITLE
Make ResponseAs[T, S] covariant on T

### ISSUE THREAD

**Issue by darl** [softwaremill/sttp#408]
I wrote following combinator:
```
def fromStatus[T, S](mapping: (Int, ResponseAs[T, S])*): ResponseAs[T, S] = {
  fromMetadata { meta =>
    val statusCode = meta.code.code
    mapping.toMap.getOrElse(statusCode, sys.error(s"No parser for status $statusCode"))
  }
}
```

And want to use it in following way:
```scala
fromStatus(
  200 -> protoBody[MyModel].map(Right(_)),
  404 -> ignore.map(_ => Left(NotFound()))
)
```

But now i forced to explicitly provide all types explicitly:
```scala
fromStatus(
  200 -> protoBody[MyModel].map(Right(_): Either[NotFound, MyModel]),
  404 -> ignore.map(_ => Left(NotFound()): Either[NotFound, MyModel])
)
```

**Comment by adamw** [comment]
Why not! Can you maybe try modifying the source (creating a PR :) ) and checking if this causes any problems?

**Comment by darl** [comment]
Yes. I will try it!

**Comment by adamw** [comment]
Fixed in 2.0.0-RC11
